app/controllers/home_controller.rb

```diff
@@ -1,16 +1,12 @@
 class HomeController < ApplicationController
   def home
-    @users = User.all
-    @projects = Project.all
-    @samples = Sample.all
     @output_data = {}
+    @samples = Sample.paginate(page: params[:page]).order('created_at DESC')
 
-    @pipeline_outputs = PipelineOutput.order("created_at DESC")
-
-    @pipeline_outputs.each do |output|
-      sample_info =  output.sample
-      project_info = output.sample.project
-      @output_data['sample_info'] = sample_info
+    @samples.each do |output|
```

Comment:
This will generate a db query for each sample. Maybe an easy way is to query once and analyze the results? 

Or do something like 
pipeline_output_ids = PipelineOutput.select("sample_id, max(id) as maxid").where(sample_id: @samples.pluck(:id)).group(:sample_id).map {|u| u.attributes }.pluck("maxid")
@pipeline_outputs = PipelineOutput.where(id: [pipeline_output_ids]) 
and analyze the results.

Reply:
@yunfangjuan I am new to this query mechanism, and just looked up `pluck`.   Is there a danger here that the output list of `maxid`s would not match the order of the input `samples.pluck(:id)`?

Reply:
you don't really need the right order. You still have to post process them into a hash to do the sample => pipeline_output mapping.